tarantool_test.go

```diff
@@ -2081,6 +2243,73 @@ func TestClientRequestObjects(t *testing.T) {
 	}
 }
 
+func TestClientRequestObjectsWithContext(t *testing.T) {
+	var err error
+	var logBuf bytes.Buffer
+	log.SetOutput(&logBuf)
+
+	conn := test_helpers.ConnectWithValidation(t, server, opts)
+	defer conn.Close()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	req := NewPingRequest().WithContext(ctx)
+	cancel()
+	resp, err := conn.Do(req).Get()
+	if err.Error() != "context is done" {
+		t.Fatalf("Failed to catch an error from done context")
+	}
+	if resp != nil {
+		t.Fatalf("Response is not nil after the occured error")
+	}
+
+	req = NewPingRequest().WithContext(nil) //nolint
+	resp, err = conn.Do(req).Get()
+	if err != nil {
+		t.Fatalf("Failed to Ping: %s", err.Error())
+	}
+	if resp == nil {
+		t.Fatalf("Response is nil after Ping")
+	}
+	if len(resp.Data) != 0 {
+		t.Errorf("Response Body len != 0")
+	}
+
+	ctx, cancel = context.WithCancel(context.Background())
+	req = NewPingRequest().WithContext(ctx)
+	fut := conn.Do(req)
+	cancel()
+	resp, err = fut.Get()
+	if resp != nil {
+		t.Fatalf("response must be nil")
+	}
+	if err == nil {
+		t.Fatalf("catched nil error")
+	}
+	if err.Error() != "context is done" {
+		t.Fatalf("wrong error catched: %v", err)
+	}
+
+	// check the output in log that the request id is unknown for the response
+	// that happens when the Future with that id was removed by the canceled context
```

Comment:
Why? It may occur when the future is removed by the contextWatchdog goroutine and after the response comes from the instance, this future can't be found int the queue.

Reply:
Unfortunately, we can't cancel a request inside the tarantool, so it is understandable that we can receive a response on the request that we have sent.

Reply:
I just want to say that this is an expected response =)

Reply:
I am agree.